### ISSUE TITLE
"IDE0074: Use compound assignment" doesn't trigger on the field keyword

### ISSUE THREAD

**Issue by MihaZupan** [dotnet/roslyn#76633]
**Version Used**:
- SDK `10.0.100-alpha.1.25056.1` (today's latest from [10.0 main](https://github.com/dotnet/sdk/blob/main/documentation/package-table.md))
- VS Version 17.13.0 Preview 2.1

**Steps to Reproduce**:

Code like this will trigger the `IDE0074: Use compound assignment` diagnostic:
```c#
private string _foo;
public string Foo
{
    get
    {
        if (_foo is null)
        {
            _foo = "foo";
        }

        return _foo;
    }
}
```

But when changed to use `field`, it won't.

```c#
public string Foo
{
    get
    {
        if (field is null)
        {
            field = "foo";
        }

        return field;
    }
}
```

**Diagnostic Id**: `IDE0074: Use compound assignment`
